ssl/statem/statem_clnt.c

```diff
@@ -160,6 +160,17 @@ static int ossl_statem_client13_read_transition(SSL *s, int mt)
             st->hand_state = TLS_ST_CR_KEY_UPDATE;
             return 1;
         }
+        if (mt == SSL3_MT_CERTIFICATE_REQUEST) {
+            if (s->post_handshake_auth == SSL_PHA_EXT_SENT) {
+                s->post_handshake_auth = SSL_PHA_REQUESTED;
+                if (!tls13_restore_handshake_digest_for_pha(s)) {
```

Comment:
Well, that's interesting. While I could put a `#error` here, would it be better to fix the original problem? Perhaps the handshake digest should be restored just before adding the message? Would it make sense to have a separate digest storage for post-auth in addition to storing the digest state through clientFinished (not sure; I didn't do it that way to avoid complicating the CertRequest/Certificate/CertVerify/Finished and the extra code to extract and generate it)?

Reply:
A thought: if KeyUpdate and NewSessionTicket are no longer added to the cache, might restoring the digest state after the client Finish also work?

Reply:
> Well, that's interesting. While I could put a #error here, would it be better to fix the original problem?

Ideally yes that would be nice (although not essential since there will no doubt be lots of other changes required for DTLS1.3)

> Perhaps the handshake digest should be restored just before adding the message?

Possibly. `ssl3_finish_mac()` could be a good place since it is called by both TLS and DTLS

> A thought: if KeyUpdate and NewSessionTicket are no longer added to the cache, might restoring the digest state after the client Finish also work?

Yes that could work. Although I think my preference would be to reset it at the beginning of a new post-handshake auth rather than after client Finish.